web/app/scripts/enforcers/enforcers-modal-controller.js

```diff
@@ -0,0 +1,94 @@
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function EnforcersModalController($log, $modalInstance, $state, $translate, $window,
+                                      $timeout, FilterState, BoundaryState, GeographyState,
+                                      RecordState, MapState) {
+
+        var maxShiftLength = Math.abs(moment.duration(24, 'hours').asHours());
+        var ctl = this;
+
+        ctl.$onInit = initialize();
+
+        function initialize() {
+            ctl.ready = false;
+            ctl.closeModal = closeModal;
+            ctl.createAssignments = createAssignments;
+            ctl.onParamChanged = onParamChanged;
+
+            RecordState.getSelected().then(function (recordType) {
+                ctl.recordType = recordType.uuid;
+            });
+
+            GeographyState.getSelected().then(function(geography) {
+                BoundaryState.getSelected().then(function (boundary) {
+                    if (boundary && boundary.data) {
+                        ctl.polygonId = boundary.uuid;
+                        /* jshint camelcase: false */
+                        ctl.polygonName = boundary.data[geography.display_field];
+                        /* jshint camelcase: true */
+                    } else {
+                        ctl.polygonId = null;
+                        ctl.polygonName = null;
+                    }
+                });
+            });
+
+            ctl.polygon = MapState.getFilterGeoJSON();
+        }
+
+        function closeModal() {
+            $modalInstance.close();
+        }
+
+        function onParamChanged() {
+            // TODO: Without this, changes to the datepicker take two digest cycles to appear on
```

Comment:
I don't think so because the other spot with a datepicker doesn't use the custom directive I made here, but it'd be a nice refactor if it did, and if we made that change then it would presumably have the same problem.